OpenRA.Game/Network/UnitOrders.cs

```diff
@@ -58,6 +58,29 @@ public static void ProcessOrder(OrderManager orderManager, World world, int clie
 						break;
 					}
 
+				case "PlaceBeacon":
+					{
+						var client = orderManager.LobbyInfo.ClientWithIndex(clientId);
+						if (client != null)
+						{
+							var player = world.FindPlayerByClient(client);
+							if (player == null)
+							{
+								if (orderManager.LocalClient.IsObserver)
+								{
+									Game.AddChatLine(Color.White, ServerChatName, "{0} placed a beacon".F(client.Name));
+									player = world.Players.Single((p) => p.PlayerName == "Everyone");
+								}
+							}
+
+							if (player != null && !player.PlayerActor.IsDead)
+								foreach (var t in player.PlayerActor.TraitsImplementing<IResolveOrder>())
```

Comment:
This offers a channel for immediate orders to bleed into world state, which can lead to desyncs.  Somebody who understands the details of our orders system really should look over this PR and suggest an alternative implementation approach.